### PR TITLE
[hive] Support recreate the hive table

### PR DESCRIPTION
**Pull request by silentxingtian**
### Purpose

Linked issue: close #xxx

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/schema/SchemaManager.java [hunk 1]
```diff
@@ -124,16 +125,28 @@ public List<Long> listAllIds() {
         }
     }
 
+    /** Check TableScheme is be modified. */
+    public void checkTableSchema(TableSchema oldSchema, TableSchema newSchema) {
+        boolean isCommon =
+                oldSchema.version() == newSchema.version()
+                        && oldSchema.id() == newSchema.id()
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/schema/SchemaManager.java:132]: I don't think we need to compare schema id and options.

**Review by JingsongLi — CHANGES_REQUESTED** [review]
Can you add test for this?

File: paimon-core/src/main/java/org/apache/paimon/schema/SchemaManager.java (hunk 1 above)
Comment (silentxingtian) [reply paimon-core/src/main/java/org/apache/paimon/schema/SchemaManager.java:132]: Okay, if there's no change in the entire logic, I'll remove the schema id and options

**Comment by silentxingtian** [comment]
> Can you add test for this?

ok,I will write a test case code and submit it again

**Pull request merged**